### ISSUE TITLE
PR merging options: merge commit, rebase and squash

### ISSUE THREAD

**Issue by bernhardmgruber** [ComputationalRadiationPhysics/contributing#47]
Previous discussion in: https://github.com/alpaka-group/alpaka/issues/1103

Should be establish a common guideline how PRs are merged? And what should it be?
Currently, alpaka does a rebase and most other projects create a merge commit.

**Comment by ax3l** [comment]
In case it's useful, I use squash commits these days on all repos, since it saves me from explaining to new contributors how to rebase :)

**Comment by psychocoderHPC** [comment]
What I do not like when using `squash commits` is that you destroy useful structured PR. If the commit is splitting up the changes into grouped commits it helps when looking later (without github) into the changes. 
In cases where the PR is very tiny (hopefully in most cases) it is very useful to squash the commits because nobody will be interested that a five-line change took N commits until the CI passed.

**Comment by SimeonEhrig** [comment]
I agree with René. A clear commit structure is a part of the documentation. That's something, which I teach my student at the moment.

**Comment by bernhardmgruber** [comment]
> If the commit is splitting up the changes into grouped commits it helps when looking later (without github) into the changes.

Fully agree.

> A clear commit structure is a part of the documentation.

Fully agree.

> In cases where the PR is very tiny (hopefully in most cases) it is very useful to squash the commits because nobody will be interested that a five-line change took N commits until the CI passed.

Here I see the PR author responsible for rebasing the commit history accordingly. But as Axel pointed out, that requires the author to be fimilar with rebasing. However, being able to rebase is such a versatile technique, I believe people should go the extra mile and acquire the skill.
